When hitting Stop during Debug, it does not always kill debug process

I am running a web server in Go (1.6) with visual studio code and the most recent plugin:

```
func main() {

    http.HandleFunc("/data", htmlHandler)

    err := http.ListenAndServe(":8080", nil)
    if err != nil {
        log.Fatal(err)
    }
    os.Exit(0)
}
```

When I hit the stop button on the small debug window, it does not kill the webserver, because after I hit F5 again I get:

>  2016/08/16 15:34:13 server.go:71: Using API v1
> 2016/08/16 15:34:13 debugger.go:61: launching process with args: [./debug]
> API server listening at: 127.0.0.1:2345
> 2016/08/16 15:34:13 debugger.go:393: continuing
> 2016/08/16 15:34:13 listen tcp :8080: bind: address already in use

So I will have to look manually for the PID and kill it. Then it works again.

@ramya-rao-a I was trying out the steps you provided. However, at `npm install` I get:

```
npm ERR! 404 Not Found
npm ERR! 404 
npm ERR! 404 'types/fs-extra' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it
npm ERR! 404 It was specified as a dependency of 'Go'
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, or http url, or git url.
```

--- a/scripts/terminateProcess.sh
+++ b/scripts/terminateProcess.sh
@@ -1,0 +1,12 @@
+#!/bin/bash
+
+terminateTree() {
+	for cpid in $(/usr/bin/pgrep -P $1); do
+		terminateTree $cpid
+	done
+	kill -9 $1 > /dev/null 2>&1
+}
+
+for pid in $*; do
+	terminateTree $pid
+done

--- a/src/debugAdapter/goDebug.ts
+++ b/src/debugAdapter/goDebug.ts
@@ -9,7 +9,7 @@ import { DebugProtocol } from 'vscode-debugprotocol';
 import { DebugSession, InitializedEvent, TerminatedEvent, ThreadEvent, StoppedEvent, OutputEvent, Thread, StackFrame, Scope, Source, Handles } from 'vscode-debugadapter';
 import { readFileSync, existsSync, lstatSync } from 'fs';
 import { basename, dirname } from 'path';
-import { spawn, ChildProcess } from 'child_process';
+import { spawn, ChildProcess, execSync, spawnSync } from 'child_process';
 import { Client, RPCConnection } from 'json-rpc2';
 import { getBinPathWithPreferredGopath } from '../goPath';
 import * as logger from 'vscode-debug-logger';
@@ -305,7 +305,7 @@ class Delve {
 				});
 			});
 		} else {
-			this.debugProcess.kill();
+			killTree(this.debugProcess.pid);
 		}
 	}
 }
@@ -755,6 +755,26 @@ class GoDebugSession extends DebugSession {
 
 function random(low: number, high: number): number {
 	return Math.floor(Math.random() * (high - low) + low);
+}
+
+function killTree(processId: number): void {
+    if (process.platform === 'win32') {
+        const TASK_KILL = 'C:\\Windows\\System32\\taskkill.exe';
+
+        // when killing a process in Windows its child processes are *not* killed but become root processes.
+        // Therefore we use TASKKILL.EXE
+        try {
+            execSync(`${TASK_KILL} /F /T /PID ${processId}`);
+        } catch (err) {
+        }
+    } else {
+        // on linux and OS X we kill all direct and indirect child processes as well
+        try {
+            const cmd = path.join(__dirname, '../../../scripts/terminateProcess.sh');
+            spawnSync(cmd, [ processId.toString() ]);
+        } catch (err) {
+        }
+    }
 }
 
 DebugSession.run(GoDebugSession);

--- a/src/debugAdapter/goDebug.ts
+++ b/src/debugAdapter/goDebug.ts
@@ -758,23 +758,23 @@ function random(low: number, high: number): number {
 }
 
 function killTree(processId: number): void {
-    if (process.platform === 'win32') {
-        const TASK_KILL = 'C:\\Windows\\System32\\taskkill.exe';
+	if (process.platform === 'win32') {
+		const TASK_KILL = 'C:\\Windows\\System32\\taskkill.exe';
 
-        // when killing a process in Windows its child processes are *not* killed but become root processes.
-        // Therefore we use TASKKILL.EXE
-        try {
-            execSync(`${TASK_KILL} /F /T /PID ${processId}`);
-        } catch (err) {
-        }
-    } else {
-        // on linux and OS X we kill all direct and indirect child processes as well
-        try {
-            const cmd = path.join(__dirname, '../../../scripts/terminateProcess.sh');
-            spawnSync(cmd, [ processId.toString() ]);
-        } catch (err) {
-        }
-    }
+		// when killing a process in Windows its child processes are *not* killed but become root processes.
+		// Therefore we use TASKKILL.EXE
+		try {
+			execSync(`${TASK_KILL} /F /T /PID ${processId}`);
+		} catch (err) {
+		}
+	} else {
+		// on linux and OS X we kill all direct and indirect child processes as well
+		try {
+			const cmd = path.join(__dirname, '../../../scripts/terminateProcess.sh');
+			spawnSync(cmd, [ processId.toString() ]);
+		} catch (err) {
+		}
+	}
 }
 
 DebugSession.run(GoDebugSession);

